Sbom tools

SebastianSchildt opened the pull request:
Refactoring, splitting the function of https://github.com/eclipse/kuksa.val/pull/756 making it pip installable

SebastianSchildt commented:
@erikbosch  there may be an error in pre-commit check?

```
Fixing sbom-tools/src/kuksa_sbom_utils/licensestore/ring.LICENSE.txt.gz
Fixing sbom-tools/src/kuksa_sbom_utils/licensestore/webpki.LICENSE.txt.gz
```

that is the "end of file" fixer, but is this is are compressed binary files, they should not be touched

SebastianSchildt commented:
Ok, it seems  pre-commit uses a python package called `identify` to check what files it is dealing with. Interestingly the results are not consistent for all our compressed files

```
(testsb-n7i5oTkr) scs2rng@RNG-C-001JT licensestore % identify-cli OpenSSL.txt.gz     
["binary", "file", "gzip", "non-executable"]
(testsb-n7i5oTkr) scs2rng@RNG-C-001JT licensestore % identify-cli ring.LICENSE.txt.gz
["binary", "file", "gzip", "non-executable", "plain-text", "text"]
```

It seems a bug to tag something as "binary" and "text", but a anyways both are tagged (correctly) as gzip type as well I just excluded that

SebastianSchildt commented:
For completeness sake raised an issue upstream https://github.com/pre-commit/identify/issues/450

erikbosch commented on sbom-tools/Readme.md:
````diff
@@ -0,0 +1,59 @@
+
+```yaml
+expressions:
+    "Apache-2.0 OR MIT": "Apache-2.0"
````
@mikehaller - this could possibly be a solution for our "unlicense" discusssion the other day

erikbosch commented on sbom-tools/src/kuksa_sbom_utils/licensestore/.DS_Store:
Isn't this a MAC-specific file? Do we really need it?

SebastianSchildt replied:
Removed

erikbosch commented on sbom-tools/LICENSE:
How is this file used; it is the same as in root of the repo right?

SebastianSchildt replied:
It is the same.
This is required, to be able to install this "folder" with pip. It seems to be required metadata.

erikbosch commented on sbom-tools/Readme.md:
Shall we call it README.md to have consistent capitalization?

SebastianSchildt replied:
Done

erikbosch commented:
Some minor comments, but I like it.

I am thinking - shall we possibly have an "example" folder? I.e. an example input file, a README that describes exactly how to run it and the expected output result. Then we could later possibly add a check in CI that running the python script still works and still produce the same result

SebastianSchildt commented:
Fixed comments. I added an example folder, but at this time opted to not check in and "Output" folder, because no tests yet, and I am not sure, whether we might still change the exact format of the output in the future.

erikbosch reviewed (approved):
LGTM
